src/session.rs

```diff
@@ -127,6 +128,14 @@ impl Session {
         }
     }
 
+    /// Get the remote static key that was possibly encrypted in the first payload
+    pub fn get_remote_static(&self) -> Option<&[u8; MAXDHLEN]> {
```

Comment:
on second peek, returning an array reference sized at MAXDHLEN is a misleading public-facing API. Might be more clean to return a `Option<&[u8]>` that's properly sliced to the length based on the chosen DH function?

Reply:
Mmm, probably. I encountered a similar issue with one of the Noise extensions I'm working on, where I end up with `[u8; MAXHASHLEN]` in the public-facing API. I haven't yet decided how to solve that one, but this API has an unambiguous meaning relative to the spec, so it should be safe to give users a trimmed slice. I'll make that change.